2003-01-17  Anton Kachalov <mouse@altlinux.org>,
	    Dmitry V. Levin <ldv@altlinux.org>

	* cmdline/apt-get.cc (TryToInstall):
	Add APT::Install::Virtual support.

--- apt-0.5.4cnc9/cmdline/apt-get.cc.orig	2003-01-17 17:11:11 +0300
+++ apt-0.5.4cnc9/cmdline/apt-get.cc	2003-01-17 17:19:10 +0300
@@ -988,7 +988,8 @@ bool TryToInstall(pkgCache::PkgIterator 
    /* This is a pure virtual package and there is a single available 
       provides */
    if (Cache[Pkg].CandidateVer == 0 && Pkg->ProvidesList != 0 &&
-       Pkg.ProvidesList()->NextProvides == 0)
+       (Pkg.ProvidesList()->NextProvides == 0 ||
+        _config->FindB("APT::Install::Virtual", false)))
    {
       pkgCache::PkgIterator Tmp = Pkg.ProvidesList().OwnerPkg();
       ioprintf(c1out,_("Note, selecting %s instead of %s\n"),
